codegenerator/cli/src/config_parsing/system_config.rs

```diff
@@ -833,34 +849,56 @@ impl Event {
     }
 
     pub fn from_evm_event_config(
-        human_cfg_event: EventConfig,
+        event_config: EvmEventConfig,
         opt_abi: &Option<EvmAbi>,
     ) -> Result<Self> {
         let event: NormalizedEthAbiEvent =
-            Event::get_abi_event(&human_cfg_event.event, opt_abi)?.into();
+            Event::get_abi_event(&event_config.event, opt_abi)?.into();
+        let topic0 = ethers::core::utils::hex::encode_prefixed(ethers::utils::keccak256(
+            event.0.abi_signature().as_bytes(),
+        ));
 
         Ok(Event {
-            name: human_cfg_event.name.unwrap_or(event.0.name.to_owned()),
+            name: event_config.name.unwrap_or(event.0.name.to_owned()),
+            payload: EventPayload::Params(event.0.inputs.clone()),
             event,
+            topic0,
         })
     }
 
-    fn get_event(&self) -> &EthAbiEvent {
-        &self.event.0
-    }
+    // TODO: Validatate that all event names are unique
+    // TODO: Validatate that event names are not reserved
+    pub fn from_fuel_event_config(
+        event_config: FuelEventConfig,
+        fuel_abi: &FuelAbi,
+    ) -> Result<Self> {
+        let event: NormalizedEthAbiEvent =
+            Event::get_abi_event(&format!("{}()", event_config.name), &None)?.into();
+
+        let log = match event_config.log_id {
+            None => {
+                let logged_type = fuel_abi.get_type_by_struct_name(event_config.name.clone()).context(
+                  "Failed to derive log ids from the event name. Use the lodId field to set it explicitely."
+                )?;
+                fuel_abi.get_log_by_type(logged_type.id)?
+            }
+            Some(log_id) => fuel_abi.get_log(&log_id)?,
+        };
 
-    pub fn get_event_inputs(&self) -> Vec<EventParam> {
-        self.get_event().inputs.clone()
+        Ok(Event {
+            name: event_config.name,
+            event,
+            payload: EventPayload::Data(log.data_type),
+            topic0: log.id,
```

Comment:
Is it appropriate to call fuel's log.id a sighash?

If so maybe we rename topic0 to sighash as a convention. Since topic0 is the sighash of an evm event.

Reply:
Done in the PR